packages/modelserver-client/src/model-server-client-api-v1.ts

```diff
@@ -0,0 +1,353 @@
+    /**
+     * Updates an existing model with the given URI with the given content in the current workspace.
+     * The return type is derived from the given typeguard. The result promise is rejected if the updated model didn't
+     * pass the typeguard check.
+     * The result promise is rejected if no matching model for the given URI exists.
+     * @param modeluri The URI of the model that should be updated.
+     * @param model The content of the updated model object either as string.
+     * @param typeGuard A guard function to check if the received model element is of the expected type.
+
+     * @returns Thew updated model as typed JSON object `M`.
+     */
+    update<M>(modeluri: string, model: string | string, typeGuard: TypeGuard<M>): Promise<M>;
+    /**
+     * Updates an existing model with the given URI with the given content string in the current workspace.
+     * A `format` string has to passed to let the modelserver know how the given content should be interpreted.
+     * @param modeluri The URI of the model that should be updated.
+     * @param model The content of the updated model object either as string.
+     * @param format The desired format.
+     * @returns Thew updated model as string.
+     */
+    update(modeluri: string, model: string, format: string): Promise<AnyObject>;
+
+    /**
+     * Persists all `dirty` changes for the model with the given URI.
+     * @param modeluri The URI of the model whose dirty changes should be saved.
+     * @returns A boolean indicating wether the saving was successful.
+     */
+    save(modeluri: string): Promise<boolean>;
+
+    /**
+     * Saves (i.e. persists the dirty changes of) all models that are loaded in the current workspace.
+     * @returns A boolean indicating wether the saving was successful.
+     */
+    saveAll(): Promise<boolean>;
+
+    /**
+     * Validates the model with the given URI and returns the {@link Diagnostic} validation result.
+     * @param modeluri The URI of the model that should be validated.
+     * @returns The validation result as {@link Diagnostic}.
+     */
+    validate(modeluri: string): Promise<Diagnostic>;
+
+    /**
+     * Retrieves the EMF validation constraints for the model with the given URI
+     * @param modeluri THe URI of the model whose constraints should be retrieved.
+     * @returns The validation constraints as string.
+     */
+    getValidationConstraints(modeluri: string): Promise<string>;
+
+    /**
+     * Retrieves the JSON schema representation of the Ecore model with the given URI.
+     * Note: currently the JSON schema can only be derived from the Ecore (i.e. Meta) model. So the given URI
+     * has to point to the Ecore model and not a concrete model instance. (e.g. `Coffee.ecore` instead of `superbrewer3000.coffee)
+     * @param modeluri The URI of the ecore model whose JSON schema should be retrieved
+     * @returns The corresponding JSON schema  as string.
```

Comment:
```suggestion
     * has to point to the Ecore model and not a concrete model instance. (e.g. `Coffee.ecore` instead of `SuperBrewer3000.coffee)
     * @param modeluri The URI of the Ecore model whose JSON schema should be retrieved
     * @returns The corresponding JSON schema as string.
```